pyproject.toml

```diff
@@ -38,23 +38,21 @@ authors = [
 ]
 
 dependencies = [
-  "astropy >= 5.3.0",
-  "matplotlib >= 3.5.0",
-  "numpy >=1.23.5",
-  "scikit-image >= 0.19.0",
-  # !=1.10.0 due to https://github.com/scipy/scipy/issues/17718
-  "scipy >= 1.9.0 , != 1.10.0",
-  "sunpy[map] >= 5.0.0",
+  "astropy >= 6.0.0",
+  "matplotlib >= 3.7.0",
+  "numpy >=1.24.0",
+  "scikit-image >= 0.21.0",
+  "scipy >= 1.11.1",
+  "sunpy[map] >= 5.1.0",
```

Comment:
[SPEC 0](https://scientific-python.org/specs/spec-0000/) recommends that packages support all major/minor versions of core dependencies that were released in the past 24 months. This mostly follows that guideline.

I updated this since this PR adds tests of the lowest allowed versions of direct dependencies.